connector/src/main/java/org/jboss/as/connector/subsystems/datasources/DataSourceDefinition.java

```diff
@@ -196,6 +196,23 @@ protected boolean isValueDiscardable(PathAddress address, String attributeName,
                 .end();
     }
 
+    static void registerTransformers130(ResourceTransformationDescriptionBuilder parentBuilder) {
+        ResourceTransformationDescriptionBuilder builder = parentBuilder.addChildResource(PATH_DATASOURCE);
+        builder.getAttributeBuilder()
+                .setDiscard(new DiscardAttributeChecker.DefaultDiscardAttributeChecker() {
+                    @Override
+                    protected boolean isValueDiscardable(PathAddress address, String attributeName, ModelNode attributeValue, TransformationContext context) {
+                        return attributeValue.equals(new ModelNode(false));
+                    }
+                }, TRACKING)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, TRACKING).end()
+                //We're rejecting operations when statistics-enabled=false, so let it through in the enable/disable ops which do not use that attribute
+                .addOperationTransformationOverride(DATASOURCE_ENABLE.getName())
+                .end()
+                .addOperationTransformationOverride(DATASOURCE_DISABLE.getName())
+                .end();
```

Comment:
Can you comment on what L209-213 are doing here and in the XA one; i.e. what problem they solve?

Looking at the impl of ResourceTransformationDescriptionBuilder and the other types it returns, these calls don't look like they would result in any change. But this is complex so I can easily be missing something.